src/AutoRest.CSharp/Common/Output/Models/MethodSignature.cs

```diff
@@ -102,4 +105,44 @@ private MethodSignature MakeSync()
             };
         }
     }
+
+    internal class MethodSignatureComparer : IEqualityComparer<MethodSignature>
+    {
+        public bool Equals(MethodSignature? x, MethodSignature? y)
+        {
+            if (ReferenceEquals(x, y))
+            {
+                return true;
+            }
+
+            if (x is null || y is null)
+            {
+                return false;
+            }
+
+            var result = x.Name == x.Name
+                && CompareType(x.ReturnType, y.ReturnType)
+                && x.Parameters.SequenceEqual(y.Parameters, new ParameterComparer());
+            return result;
+        }
+
+        public int GetHashCode([DisallowNull] MethodSignature obj)
+        {
+            return HashCode.Combine(obj.Name, obj.ReturnType);
+        }
+
+        private bool CompareType(CSharpType? x, CSharpType? y)
```

Comment:
Can this method be moved to `CSharType`?

Reply:
[EqualsTo ](https://github.com/Azure/autorest.csharp/blob/2ea857bbb2a719d7a1790dd501c856ba38de32eb/src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs#L96) has a different implementation with its own purpose.

Reply:
Moved it to `TypeExtensions.cs`.